force-app/src/classes/Argument.cls

```diff
@@ -127,7 +127,7 @@ global class Argument {
     }
 
     public Boolean matches(final Object callArgument) {
-      return callArgument == this.callArgumentToMatch;
+      return String.valueOf(callArgument) == this.toString();
     }
 
     override public String toString() {
```

Comment:
**suggestion:** boy scout

it should return `String.valueOf(this.callArgumentToMatch);`